domain/secretbackend/state/state.go

```diff
@@ -361,8 +427,19 @@ func (s *State) ListSecretBackendsForModel(ctx context.Context, modelUUID coremo
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	// TODO: implement for inUse. JUJU-5707
-	stmt, err := s.Prepare(`
+
+	inUseCondition := ""
+	var inputArgs []any
+	if !includeEmpty {
+		inputArgs = append(inputArgs, SecretBackendReference{ModelID: modelUUID})
+		inUseCondition = fmt.Sprintf(`
+WHERE EXISTS (
+    SELECT 1
+    FROM secret_backend_reference
+    WHERE secret_backend_uuid = b.uuid AND model_uuid = $SecretBackendReference.model_uuid
+) OR b.name = '%s' OR b.name = '%s'`, kubernetes.BackendName, juju.BackendName)
```

Comment:
This would be more efficient as a join right?
Have you done query plan to check how well this performs?